src/combobox/combobox.component.ts

```diff
@@ -454,10 +454,12 @@ export class ComboBox implements OnChanges, AfterViewInit, AfterContentInit, OnD
 	ngOnChanges(changes) {
 		if (changes.items) {
 			this.view.items = changes.items.currentValue;
-			// If new items are added into the combobox while there is search input,
-			// repeat the search.
-			this.onSearch(this.input.nativeElement.value, false);
 			this.updateSelected();
+			// If new items are added into the combobox while there is search input,
+			// repeat the search. Search should only trigger for type 'single' when there is no value selected.
+			if ( this.type === "multi" || (this.type === "single" && !this.selectedValue)) {
+				this.onSearch(this.input.nativeElement.value, false);
+			}
```

Comment:
Should `updateSelected()` be called before `onSearch` in this case? Would updating the selected value after new items are added would make `this.selectedValue` always false, and make `onSearch` always trigger?

Reply:
Thanks for looking at this @Donisius 

The proposed change has the conditional `onSearch` moved after `updateSelected()` so that `this.selectedValue` is updated before the search is called - is that what you mean?

Reply:
@fionamurphy Yes, testing it locally with single select still triggered search when new items were added and a selected value was present. This is the example I used to test it:

```
@Component({
	selector: "app-dynamic-list-combobox",
	template: `
		<ibm-combo-box
			[(items)]="items"
			type="single">
			<ibm-dropdown-list></ibm-dropdown-list>
		</ibm-combo-box>
	`
})
class DynamicListComboBox implements AfterViewInit {
	items = [
		{
			content: "one"
		},
		{
			content: "two"
		},
		{
			content: "three"
		},
		{
			content: "four"
		}
	];

	ngAfterViewInit() {
		setInterval(() => {
			const newItems = JSON.parse(JSON.stringify(this.items));
			newItems.push({ content: `New ${newItems.length}` });
			this.items = newItems;
		}, 4000);
	}
}
```

Reply:
@Donisius `this.updateSelected()` is getting the selected value from `dropdown-list.component.ts`  `getSelected()`

```
getSelected(): ListItem[] {
		let selected = this.getListItems().filter(item => item.selected);
		if (selected.length === 0) {
			return [];
		}
		return selected;
	}
```

If no item has a truthy selected value this.selectedValue won't be set and the search is triggered.
I guess your example would work if the selected property was set on one of the items.

Reply:
@fionamurphy Okay, that makes sense to me!